aiida_siesta/calculations/tkdict.py

```diff
@@ -110,6 +81,80 @@ def __repr__(self):
     def __str__(self):
         return self._storage.__str__()
 
+    def values(self):
+        """
+        Return list of values.
+
+        """
+        return [self[k] for k in self]
+
+    def untranslated_keys(self):
+        """
+        Return a list of last occurencies of the untranslated keys. For each translated_key,
+        it is stored as second entry of the self._storage[translated_key] tuple.
+        """
+        return [self._storage[k][1] for k in self._storage]
+
+    def keys(self):
+        """
+        Return list of translated keys. The self_storage keys are already translated.
+
+        """
+        return self._storage.keys()
+
+    def untranslated_items(self):
+        """
+        Return a list of the items with the last occurencies of the untranslated keys
+        as key.
+        """
+        return [(self._storage[key][1], self._storage[key][0]) for key in self]
+
+    def items(self):
+        """
+        Return a list of the items with the translated key as key.
+
+        """
+        return [(key, self._storage[key][0]) for key in self]
+
+    def translated_dict(self):
+        """
+        Return a dictionary, where the key are the translated keys
+        """
+        new_dict = self._storage.copy()
```

Comment:
Why do you need to copy the storage here, instead of just `new_dict = {}`? :thinking:

Reply:
Just to reuse the keys of self._storage, but it's the same. Why do you think is better to have a new_dict??

Reply:
Just to save time, and because it doesn't have any use. At the end you are going to overwrite it all.

In fact, this would be the perfect place for a dict comprehension, wouldn't it? As you've done in the methods that return lists.

```python
def translated_dict(self):
    return {key: val[0] for key, val in self._storage.items()}
```

Reply:
I agree with @pfebrer96, comprehension suits better here. Not maybe because of the speed, but it is more expressive. The result is the same.